### PR TITLE
Fix 'run' behaviour with networks

### PR DESCRIPTION
**Pull request by aanand**
- Test that one-off containers join all networks
- Don't set any aliases

### CODE REVIEW

**Comment by dnephin** [comment]
LGTM

**Comment by aanand** [comment]
Updated - we were relying on a non-feature of Engine that's about to get fixed. Proper fix is now in, and reliant on https://github.com/d11wtq/dockerpty/pull/54.

**Comment by aanand** [comment]
https://github.com/d11wtq/dockerpty/pull/54 is merged.

File: requirements.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
 PyYAML==3.11
 cached-property==1.2.0
-dockerpty==0.3.4
 docopt==0.6.1
 enum34==1.0.4
+git+https://github.com/d11wtq/dockerpty.git@master#egg=dockerpty
```
Comment (dnephin) [inline requirements.txt]: I think we should be using a sha here, otherwise it could break without any changes to compose

**Comment by mavenugo** [comment]
@aanand FYI. If a container needs to be connected to multiple networks, the recommended approach is -
1. create the container in 1 network with the required options
2. connect to other networks with its own network specific option
3. start the container

Though we allow connecting to networks after start, the application would assume that all the networks are connected to when it starts. Hence, it is better to follow the above sequence.

**Comment by dnephin** [comment]
I think that's what we're doing now.

Got a link to the engine PR?

Other than the pinning `master`, LGTM

File: requirements.txt (hunk 1 above)
Comment (aanand) [inline requirements.txt]: True - same goes for docker-py, I'll fix that separately

**Comment by aanand** [comment]
Cherry-picked into #2753

**Pull request merged**